search-provider/eks-discovery-feed-provider.c

```diff
@@ -458,12 +461,15 @@ article_card_descriptions_cb (GObject *source,
 
   GVariantBuilder builder;
   g_variant_builder_init (&builder, G_VARIANT_TYPE ("aa{ss}"));
-  for (GSList *l = models; l; l = l->next)
+
+  for (i = 0; i < MAX_KNOWLEDGE_APPS; i++)
     {
       /* Start building up object */
       g_variant_builder_open (&builder, G_VARIANT_TYPE ("a{ss}"));
 
-      EkncContentObjectModel *model = l->data;
+      gint index = (get_day_of_year () + i) % g_slist_length (models);
+      EkncContentObjectModel *model = g_slist_nth (models, index)->data;
```

Comment:
Another way you could do this without converting the data structure is to build up a start and end index for the models that you're going to fetch out of the list outside the loop, then iterate over all the models and add the ones which are within that range.